### ISSUE TITLE
Cross validation & shuffling

### ISSUE THREAD

**Issue by Karlheinzniebuhr** [Wensi-Tang/OS-CNN#9]
How does cross-validation happen in OS-CNN? Does it apply shuffling to the testset? And if not, how can I prevent overfitting?

**Comment by Wensi-Tang** [comment]
1) How does cross-validation happen in OS-CNN?
The code does not have any cross-validation here.

2) Does it apply shuffling to the testset? 
No, it is just trained on the X_train and y_train and tested with X_test and y_test.

3) And if not, how can I prevent overfitting? 
The code is not like those methods in [Scikit-learn](https://scikit-learn.org/stable/) which are ready to use. The code aims to show that, for 1D-CNN, we could build a network that covers all reception fields instead of searching for the specific kernel configuration. To compare with others, the results that we report for each dataset all use the same hyperparameters. The hyperparameters setting detail can be found here [link](https://github.com/Wensi-Tang/OS-CNN/blob/master/Classifiers/OS_CNN/OS_CNN_easy_use.py). 

If you want to use the model to achieve high performance, there is still some work to do. You might need to tune hyperparameters and use some techniques such as ensemble and even changing loss. 

To do that, I would suggest starting by reading the [fit function](https://github.com/Wensi-Tang/OS-CNN/blob/cb90c29735c345043c17806346ba11cc101b9cb4/Classifiers/OS_CNN/OS_CNN_easy_use.py#L54) which has 79 lines and no complex code in it. The model is initialized [here](https://github.com/Wensi-Tang/OS-CNN/blob/cb90c29735c345043c17806346ba11cc101b9cb4/Classifiers/OS_CNN/OS_CNN_easy_use.py#L88) and trained [here](https://github.com/Wensi-Tang/OS-CNN/blob/cb90c29735c345043c17806346ba11cc101b9cb4/Classifiers/OS_CNN/OS_CNN_easy_use.py#L109)

If you want to change the structure of OS-CNN, you could use different 
```Python
layer_parameter_list
```
in 
```Python
torch_OS_CNN = OS_CNN(layer_parameter_list, n_class.item(), False).to(self.device)
```
the layer_parameter_list  follows the pattern below to set
```Python
layer_parameter_list = [
[(number_of_input_channel, number_of_output_channel, kernel_size),(),(),....],  # layer 1
[(),(),(),...],   # layer 2
[(),(),(),...],   # layer 3
.....             # layer ...
]
```

 More details of layer_parameter_list  can be found [here](https://github.com/Wensi-Tang/OS-CNN/blob/master/Appendix/OS-CNN_network_structure.ipynb)

**Comment by Karlheinzniebuhr** [comment]
Thank you that helps
